actions/node_create_snmpv3.py

```diff
@@ -0,0 +1,136 @@
+# Licensed to the StackStorm, Inc ('StackStorm') under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import re
+
+from lib.actions import OrionBaseAction
+from lib.utils import send_user_error
+
+
+class NodeCreateSNMPv3(OrionBaseAction):
+    def run(self,
+            node,
+            ip_address,
+            poller,
+            snmpv3_username,
+            privacy_protocol,
+            privacy_password,
+            auth_protocol,
+            auth_password,
+            status):
+        """
+        Create an node in Orion.
+        """
+        results = {}
+
+        results['label'] = self.connect()
+        self.logger.info("Connecting to Orion: {}".format(
+            results['label']))
+
+        orion_node = self.get_node(node)
+        if orion_node.npm:
+            self.logger.error(
+                "Node ({}) already in Orion: {}".format(orion_node,
+                                                        results['label']))
+
+            send_user_error("Node and/or IP is already in Orion!")
+            raise ValueError("Node and/or IP already exists!")
+
+        orion_ip_address = self.get_node(ip_address)
+        if orion_ip_address.npm:
+            self.logger.error(
+                "IP ({}) already in Orion: {}".format(
+                    orion_ip_address,
+                    results['label']))
+
+            send_user_error("IP is already in Orion!")
+            raise ValueError("IP already exists!")
+
+        self.logger.info(
+            "Checking node ({}) is not in Orion: {}".format(
+                node,
+                results['label']))
+
+        # engineID if happens to be None, default to the primary.
+        if poller is not None:
+            engineID = self.get_engine_id(poller)
+        else:
+            engineID = 1
```

Comment:
I likely wrote the above code in `NodeCreate()`, the something like this would be cleaner and easier to understand.

```suggestion
        engineID = 1
        if poller:
            engineID = self.get_engine_id(poller)
```

Reply:
Looking at the action config, `poller` defaults to `primary` and these lines 

https://github.com/StackStorm-Exchange/stackstorm-orion/blob/79fa88a33e4b1959187877fde357042396bb2c96/actions/lib/actions.py#L237-L238

Will map `poller` to the correct value. So the if statement and else are not required.

Reply:
I agree this is a lot of duplicated code. Wondering if there is a way we could put this in a generic action and only add the specific stuff in this action and in the node create action.

Reply:
I have updated the poller selection logic as suggested above.  I can look to add the SNMPv3 options to the general Node_Create action but I now have many workflows in place that reference my v3 action by name so it will be a long time and many other edits before I will move be able to support my code a consolidated node create action.